### PR TITLE
Network modelling prototype 1

### PR DESCRIPTION
**Pull request by rzietal**

### CODE REVIEW

**Comment by DhairyaLGandhi** [comment]
Most of it looks decent, there are a few places where the formatting needs to be fixed however.

File: src/model.jl [hunk 1]
```diff
@@ -82,7 +89,8 @@ function (u::Unet)(x::AbstractArray)
   up_x2 = u.up_blocks[2](up_x1, x2)
   up_x3 = u.up_blocks[3](up_x2, x1)
   up_x5 = u.up_blocks[4](up_x3, op)
-  tanh.(u.up_blocks[end](up_x5))
+  u.up_blocks[end](up_x5)
```
Comment (DhairyaLGandhi) [inline src/model.jl]: Why remove the activation here? If would be better to hold it as a parameter to `Unet` that can be set by the user and let the default be `tanh`. For anyone who doesn't need it, they can set it to `identity`

File: test/runtests.jl [hunk 2]
```diff
@@ -1,35 +1,59 @@
 using Test, UNet
 using UNet.Flux, UNet.Flux.Zygote
+using Base.Iterators
+using FileIO
+using Images
+using JLD
```
Comment (DhairyaLGandhi) [inline test/runtests.jl:6]: This needs to be added to Project.toml under the test dependencies

File: test/runtests.jl [hunk 3]
```diff
@@ -1,35 +1,59 @@
...
-  broken_ip = rand(Float32, 299, 299, 1, 1)
-  @test_throws DimensionMismatch size(u(broken_ip)) == size(broken_ip)
-end
-
-@testset "Gradient Tests" begin
-  u = Unet()
-  ip = rand(Float32, 256, 256, 1,1)
-  gs = gradient(Flux.params(u)) do
-    sum(u(ip))
+  nEpochs = 1000
```
Comment (DhairyaLGandhi) [inline test/runtests.jl:30]: This would be very slow to run on CI

**Review by DhairyaLGandhi — COMMENTED** [review]
Thanks for looking into this!

File: src/model.jl (hunk 1 above)
Comment (rzietal) [reply src/model.jl]: As far as I know there is no tanh activation function in the last UNET layer only softmax (see sect 3 of the paper). Then, Flux recommends using logitcrossentropy for numerical stability instead of cross entropy and softmax separated hence I removed softmax altogether as it is included in the loss function itself.

File: test/runtests.jl (hunk 3 above)
Comment (rzietal) [reply test/runtests.jl:30]: What's CI?

File: src/model.jl (hunk 1 above)
Comment (DhairyaLGandhi) [reply src/model.jl]: In that case, I think it would be nice if we could have the default be identity and have a parameter in Unet that can control the activation

File: test/runtests.jl (hunk 3 above)
Comment (DhairyaLGandhi) [reply test/runtests.jl:30]: Continuous integration - it runs the test suite with every commit as you can see we do ours on GitHub actions
